code/modules/client/client_defines.dm

```diff
@@ -1,8 +1,38 @@
-
+/**
+ * Client datum
+ *
+ * A datum that is created whenever a user joins a BYOND world, one will exist for every active connected
+ * player
+ *
+ * when they first connect, this client object is created and [/client/New] is called
+ *
+ * When they disconnect, this client object is deleted and [/client/Del] is called
+ *
+ * All client topic calls go through [/client/Topic] first, so a lot of our specialised
+ * topic handling starts here
+ */
 /client
-		//////////////////////
-		//BLACK MAGIC THINGS//
-		//////////////////////
+
+	/**
+	 * This line makes clients parent type be a datum
+	 *
+	 * By default in byond if you define a proc on datums, that proc will exist on nearly every single type
+	 * from icons to images to atoms to mobs to objs to turfs to areas, it won't however, appear on client
+	 *
+	 * instead by default they act like their own independent type so while you can do istype(icon, /datum)
+	 * and have it return true, you can't do istype(client, /datum), it will always return false.
+	 *
+	 * This makes writing oo code hard, when you have to consider this extra special case
```

Comment:
totally nitpicking here but including "oop" instead of "oo" may help people not miss the acronym
```suggestion
	 * This makes writing oop code hard, when you have to consider this extra special case
```

Reply:
also maybe capitalize it

Reply:
that would make it
This makes writing object oriented programming code hard

which just isn't gramatically correct.